src/dialogs/quick-bar/ha-quick-bar.ts

```diff
@@ -43,154 +51,224 @@ export class QuickBar extends LitElement {
 
   @internalProperty() private _entityItems: QuickBarItem[] = [];
 
-  @internalProperty() private _items: QuickBarItem[] = [];
+  @internalProperty() private _items?: QuickBarItem[] = [];
 
-  @internalProperty() private _itemFilter = "";
+  @internalProperty() private _filter = "";
 
   @internalProperty() private _opened = false;
 
   @internalProperty() private _commandMode = false;
 
   @internalProperty() private _commandTriggered = -1;
 
-  @internalProperty() private _activatedIndex = 0;
+  @internalProperty() private _done = false;
 
-  @query("paper-input", false) private _filterInputField?: HTMLElement;
+  @query("search-input", false) private _filterInputField?: HTMLElement;
 
-  @query("mwc-list-item", false) private _firstListItem?: HTMLElement;
+  private _focusSet = false;
 
   public async showDialog(params: QuickBarParams) {
     this._commandMode = params.commandMode || false;
-    this._opened = true;
     this._commandItems = this._generateCommandItems();
     this._entityItems = this._generateEntityItems();
+    this._opened = true;
   }
 
   public closeDialog() {
     this._opened = false;
-    this._itemFilter = "";
+    this._done = false;
+    this._focusSet = false;
+    this._filter = "";
     this._commandTriggered = -1;
-    this._resetActivatedIndex();
+    this._items = [];
     fireEvent(this, "dialog-closed", { dialog: this.localName });
   }
 
+  protected updated(changedProperties: PropertyValues) {
+    if (
+      this._opened &&
+      (changedProperties.has("_filter") ||
+        changedProperties.has("_commandMode"))
+    ) {
+      this._setFilteredItems();
+    }
+  }
+
   protected render() {
     if (!this._opened) {
       return html``;
     }
 
     return html`
-      <ha-dialog .heading=${true} open @closed=${this.closeDialog} hideActions>
-        <paper-input
+      <ha-dialog
+        .heading=${true}
+        open
+        @opened=${this._handleOpened}
+        @closed=${this.closeDialog}
+        hideActions
+      >
+        <search-input
           dialogInitialFocus
           no-label-float
           slot="heading"
           class="heading"
-          @value-changed=${this._entityFilterChanged}
+          @value-changed=${this._handleSearchChange}
           .label=${this.hass.localize(
             "ui.dialogs.quick-bar.filter_placeholder"
           )}
-          type="search"
-          value=${this._commandMode ? `>${this._itemFilter}` : this._itemFilter}
+          .filter=${this._commandMode ? `>${this._filter}` : this._filter}
           @keydown=${this._handleInputKeyDown}
-          @focus=${this._resetActivatedIndex}
```

Comment:
Why was all the activated index stuff removed in a perf PR?

Reply:
Because it didn't work. Also it was using the wrong methods. If we would want to make the focus color different, we should change the css for that.

Reply:
Fair enough!